### PR TITLE
python: fix indents

### PR DESCRIPTION
**Pull request by georgeharker**
Python will fail to indent after a for statement

eg

```
for i in range(5):<CR>
*
Cursor incorrectly here
```

will not indent.  Same for if statements.

```
if True:
*
Cursor incorrectly here
```

This is due to the line which validates if the node starts and ends on the same line - the implications of changing this for all languages are unclear.  However, for python an additional tag "indent_immediate" is set which allows these statements to be indented passing them through to the remainder of the indentation logic below.

### CODE REVIEW

**Comment by theHamsta** [comment]
Amazing! This seems to fix a lot of things. Could you add few tests for the improvements? It's just putting test files here: https://github.com/nvim-treesitter/nvim-treesitter/blob/350e7e5e31a5265fbccdf96b00a413c32583c2cc/tests/indent/python_spec.lua#L4 and specifying where to add a new line: https://github.com/nvim-treesitter/nvim-treesitter/blob/350e7e5e31a5265fbccdf96b00a413c32583c2cc/tests/indent/python_spec.lua#L19-L30

**Comment by theHamsta** [comment]
This could potentially also apply to other indent-based languages like toml.

**Comment by georgeharker** [comment]
How do I run the tests so I can iterate locally?  Did the various searches and couldn't find info.

**Comment by georgeharker** [comment]
still not sure how to run locally, but added some tests, and fixed up a bare try: which is error status, but should indent for ease

**Comment by amaanq** [comment]
check ci logs, 1 test failed

**Comment by georgeharker** [comment]
Have some fixes for the way aligned_indent worked, so that it explicitly checks if the last line needs additional indent (controlled by a final_line_indent metadata setting

this handles

```
if (blah
        and needs_more_indent):
    body
````

and

```
while (blah
       and doesn't_need_more_indent):
    body
````

**Comment by georgeharker** [comment]
Will rebase later today

File: queries/python/indents.scm [hunk 1]
```diff
@@ -21,24 +16,45 @@
...
+  condition: (parenthesized_expression) @aligned_indent
+  (#set! "delimiter" "()")
+  (#set! "final_line_indent" 1) ; parenthesized_expression already indented
 )
+
 ((ERROR "(" . (_)) @aligned_indent
  (#set! "delimiter" "()"))
 ((argument_list) @aligned_indent
  (#set! "delimiter" "()"))
-((argument_list) @aligned_indent
```
Comment (theHamsta) [inline queries/python/indents.scm:38]: Can you explain why this was a problem?

**Comment by theHamsta** [comment]
Hmm, again this. Rebasing would close the PR...

**Comment by georgeharker** [comment]
Yep. I'll do that later today for sure.

**Pull request closed without merging**